### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-docker.git`](https://github.com/cisagov/skeleton-docker.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/domain-manager-ui.git domain-manager-ui
    cd domain-manager-ui
    git remote add skeleton https://github.com/cisagov/skeleton-docker.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .pre-commit-config.yaml README.md docker-compose.yml tests/container_test.py 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may *append* to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

---

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -11,6 +16,18 @@ updates:
...
+      - dependency-name: actions/checkout
+      - dependency-name: actions/setup-python
+      # Managed by cisagov/skeleton-docker
+      # - dependency-name: actions/download-artifact
+      # - dependency-name: actions/github-script
+      # - dependency-name: actions/upload-artifact
+      # - dependency-name: docker/build-push-action
+      # - dependency-name: docker/login-action
+      # - dependency-name: docker/setup-buildx-action
+      # - dependency-name: docker/setup-qemu-action
```
Comment (jsf9k) [inline .github/dependabot.yml]: These lines should be uncommented.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -4,12 +4,17 @@ name: build
...
+      - "v*.*.*"
   pull_request:
   schedule:
-    - cron: '0 10 * * *'  # everyday at 10am
+<<<<<<< HEAD
+    # everyday at 10am
+    - cron: "0 10 * * *"
+=======
+    - cron: "0 10 * * *"  # everyday at 10am
+>>>>>>> 9cea64da4a1379cdde3c509a09fc548f0da8b034
```
Comment (jsf9k) [inline .github/workflows/build.yml]: You have committed a merge conflict.  Please resolve this.

File: README.md [hunk 3]
```diff
@@ -52,23 +52,68 @@ docker run cisagov/domain-manager-ui:1.0.0
...
+        ports:
+          - target: 8080
+            published: 8080
+            protocol: tcp
+        secrets:
+          - source: quote_txt
+            target: quote.txt
+    ```
+
+>>>>>>> 9cea64da4a1379cdde3c509a09fc548f0da8b034
```
Comment (jsf9k) [inline README.md]: You have committed a merge conflict. Please resolve.

File: docker-compose.yml [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
+<<<<<<< HEAD
+=======
+
+# This Docker composition file is used to build and test the container
+
+secrets:
+  quote_txt:
+    file: ./src/secrets/quote.txt
+
+>>>>>>> 9cea64da4a1379cdde3c509a09fc548f0da8b034
```
Comment (jsf9k) [inline docker-compose.yml]: You have committed a merge conflict.  Please resolve.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,14 +1,33 @@
...
     image: cisagov/dm-ui
     restart: unless-stopped
+=======
+      dockerfile: Dockerfile
+    image: cisagov/example
+    init: true
+    restart: "no"
+    environment:
+      - ECHO_MESSAGE=Hello World from docker compose!
+>>>>>>> 9cea64da4a1379cdde3c509a09fc548f0da8b034
```
Comment (jsf9k) [inline docker-compose.yml]: You have committed a merge conflict.  Please resolve.

File: tests/container_test.py [hunk 6]
```diff
@@ -5,7 +5,19 @@
...
+# Third-Party Libraries
+import pytest
+
+ENV_VAR = "ECHO_MESSAGE"
+ENV_VAR_VAL = "Hello World from docker compose!"
+READY_MESSAGE = "This is a debug message"
+SECRET_QUOTE = (
+    "There are no secrets better kept than the secrets everybody guesses."  # nosec
+)
+>>>>>>> 9cea64da4a1379cdde3c509a09fc548f0da8b034
```
Comment (jsf9k) [inline tests/container_test.py]: You have committed a merge conflict.  Please resolve.

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -80,22 +87,15 @@ repos:
...
           - --config=.bandit.yml
-  # Run bandit everything but tests directory
-  - repo: https://github.com/PyCQA/bandit
-    rev: 1.7.0
-    hooks:
-      - id: bandit
-        name: bandit (everything else)
-        exclude: tests
-  - repo: https://github.com/python/black
-    rev: 21.12b0
```
Comment (jsf9k) [inline .pre-commit-config.yaml]: Why was this block removed?

**Review by jsf9k — CHANGES_REQUESTED** [review]
There are a lot of extant merge conflicts.

File: tests/container_test.py (hunk 6 above)
Comment (itsmostafa) [reply tests/container_test.py]: @jsf9k resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: docker-compose.yml (hunk 5 above)
Comment (itsmostafa) [reply docker-compose.yml]: @jsf9k resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: docker-compose.yml (hunk 4 above)
Comment (itsmostafa) [reply docker-compose.yml]: resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: README.md (hunk 3 above)
Comment (itsmostafa) [reply README.md]: resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: .github/workflows/build.yml (hunk 2 above)
Comment (itsmostafa) [reply .github/workflows/build.yml]: resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: .github/dependabot.yml (hunk 1 above)
Comment (itsmostafa) [reply .github/dependabot.yml]: resolved with [d38e133](https://github.com/cisagov/domain-manager-ui/pull/247/commits/d38e1335f747335baa33f87d2745c9cf0cc84808)

File: .pre-commit-config.yaml (hunk 7 above)
Comment (itsmostafa) [reply .pre-commit-config.yaml]: resolved with [08acc63](https://github.com/cisagov/domain-manager-ui/pull/247/commits/08acc63eebd75bfe77a1b1058cdad6e544ba5677)

**Comment by jsf9k** [comment]
It looks like the linting stage is failing because of some prettification in `src/domainManagementUI/src/app/app.component.spec.ts`.  You'll need to get the linter to pass before I can approve the PR.

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -277,36 +278,37 @@ jobs:
           context: .
           file: ./Dockerfile
           outputs: type=docker,dest=dist/image.tar
-          tags: ${{ env.IMAGE_NAME }}:latest  # not to be pushed
+          # not to be pushed
+          tags: ${{ env.IMAGE_NAME }}:latest
```
Comment (dav3r) [inline .github/workflows/build.yml]: We prefer to avoid making unnecessary changes from the [skeleton](https://github.com/cisagov/skeleton-docker/blob/9cea64da4a1379cdde3c509a09fc548f0da8b034/.github/workflows/build.yml#L264):
```suggestion
          tags: ${{ env.IMAGE_NAME }}:latest # not to be pushed
```

The fewer downstream changes from the skeleton, the simpler it is to merge in new changes from the skeleton.

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -277,36 +278,37 @@ jobs:
...
-              fromJson(needs.prepare.outputs.repometa).name }}
+            fromJson(needs.prepare.outputs.repometa).name }}
 
             org.opencontainers.image.url=${{
-              fromJson(needs.prepare.outputs.repometa).html_url }}
+            fromJson(needs.prepare.outputs.repometa).html_url }}
 
             org.opencontainers.image.version=${{
-              needs.prepare.outputs.source_version }}"
+            needs.prepare.outputs.source_version }}"
```
Comment (dav3r) [inline .github/workflows/build.yml]: Same as above regarding unnecessary changes from the skeleton- a bit annoying in cases like this since we all use different editors that may auto-adjust spacing, but if you catch it before you commit them, it's not too bad.

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -415,27 +417,27 @@ jobs:
...
```
Comment (dav3r) [inline .github/workflows/build.yml]: Same as above.

**Review by dav3r — APPROVED** [review]
Looks great, except for a few spacing changes that should be put back to as they were in the skeleton. 👍

**Review by jsf9k — APPROVED** [review]
Thanks for taking care of the issues I mentioned!

Please address @dav3r's issues before merging.

File: .github/workflows/build.yml (hunk 8 above)
Comment (itsmostafa) [reply .github/workflows/build.yml]: @dav3r i tried making this change before and i get this pre-commit error:
```bash
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 2

.github/workflows/build.yml
  281:46    warning  too few spaces before comment  (comments)
  ```

Comment (itsmostafa) [reply .github/workflows/build.yml]: I resolved it by adding 2 spaces between the comment and the command instead of 1: [3b80f19](https://github.com/cisagov/domain-manager-ui/pull/247/commits/3b80f194568e20eed6de5b5d608b0095b93aa792)

File: .github/workflows/build.yml (hunk 10 above)
Comment (itsmostafa) [reply .github/workflows/build.yml]: @dav3r resolved with [69d857f](https://github.com/cisagov/domain-manager-ui/pull/247/commits/69d857f737deb5c8e8bc0dae12f6be7aeb175a6e)

File: .github/workflows/build.yml (hunk 8 above)
Comment (dav3r) [reply .github/workflows/build.yml]: Ah, sorry, that was my bad on the suggested commit - thanks for fixing that.

File: .github/workflows/build.yml (hunk 9 above)
Comment (dav3r) [reply .github/workflows/build.yml]: Fixed in https://github.com/cisagov/domain-manager-ui/pull/247/commits/a1d42e74984490fdf467f0f56f7d83180f40f24c 👍

**Review by itsmostafa — APPROVED** [review]
